lib/core/src/sdk.rs

```diff
@@ -1950,7 +1988,7 @@ mod tests {
                 for status in &unrecoverable_states {
                     let persisted_swap = trigger_swap_update!(
                         "chain",
-                        Some(direction),
+                        NewSwapArgs::default().set_direction(direction),
```

Comment:
Bellow ( on line 1999) we compare to either failed or refundable. I think it s better to split then according to the swap type and compare to a specific value.